plugins/catalog-react/src/components/EntityPeekAheadPopover/EntityPeekAheadPopover.tsx

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { entityRouteRef } from '../../routes';
+import useAsyncFn from 'react-use/lib/useAsyncFn';
+import { catalogApiRef } from '../../api';
+import React, { PropsWithChildren, useEffect } from 'react';
+import HoverPopover from 'material-ui-popup-state/HoverPopover';
+import {
+  bindHover,
+  bindPopover,
+  usePopupState,
+} from 'material-ui-popup-state/hooks';
+import {
+  Box,
+  Button,
+  Card,
+  CardActions,
+  CardContent,
+  Chip,
+  makeStyles,
+  Tooltip,
+  Typography,
+} from '@material-ui/core';
+import { Alert, Skeleton } from '@material-ui/lab';
+import EmailIcon from '@material-ui/icons/Email';
+import InfoIcon from '@material-ui/icons/Info';
+import { useApiHolder, useRouteRef } from '@backstage/core-plugin-api';
+import {
+  CompoundEntityRef,
+  isUserEntity,
+  isGroupEntity,
+  UserEntity,
+  GroupEntity,
+  Entity,
+} from '@backstage/catalog-model';
+import { Link, Progress } from '@backstage/core-components';
+
+/**
+ * Properties for an entity popover on hover of a component.
+ *
+ * @public
+ */
+export type EntityPeekAheadPopoverProps = PropsWithChildren<{
+  entityRef: CompoundEntityRef;
```

Comment:
Components like this generally always communicate using string form entity refs. So it would be good to change this to a string.

I see you're rendering parts of it below. You could still parseEntityRef in the component, and render those ONLY while the entity itself hasn't loaded yet. Because after it's loaded, you should take the kind, namespace and name out of the entity instead to make sure that you get all of the right casing.

Alternatively, rendering nothing at all (except the progress bar) until you have the entity loaded, to make it simpler.

Reply:
done.